src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2574,6 +2578,49 @@ MavlinkReceiver::handle_message_onboard_computer_status(mavlink_message_t *msg)
 	_onboard_computer_status_pub.publish(onboard_computer_status_topic);
 }
 
+void MavlinkReceiver::handle_message_statustext(mavlink_message_t *msg)
+{
+	if (msg->sysid == mavlink_system.sysid) {
+		// log message from the same system
+
+		mavlink_statustext_t statustext;
+		mavlink_msg_statustext_decode(msg, &statustext);
+
+		struct log_message_s log_message;
+
+		switch (statustext.severity) {
+			case MAV_SEVERITY_EMERGENCY:
+			case MAV_SEVERITY_ALERT:
+			case MAV_SEVERITY_CRITICAL:
+				log_message.severity = 0;
+				break;
+
+			case MAV_SEVERITY_ERROR:
+				log_message.severity = 3;
+				break;
+
+			case MAV_SEVERITY_WARNING:
+				log_message.severity = 4;
+				break;
+
+			case MAV_SEVERITY_NOTICE:
+			case MAV_SEVERITY_INFO:
+				log_message.severity = 6;
+				break;
+
+			default:
+				return;
+		}
+
+		log_message.timestamp = hrt_absolute_time();
+
+		strncpy((char *)log_message.text, statustext.text, sizeof(log_message.text));
+		log_message.text[sizeof(log_message.text) - 1] = 0; // ensure 0-termination
```

Comment:
You should null terminate the end of the STATUSTEXT string, which is shorted than the log_message max.